docs/blog/posts/introduction-to-bpfd.md

```diff
@@ -0,0 +1,611 @@
+
+Another key functional advantage that bpfd offers over libbpf or the Cilium
+ebpf-go library is support for multiple XDP programs. Standard XDP only allows a
+single XDP program on a given interface, while bpfd supports loading multiple
+XDP programs on each interface using the [multi-prog] protocol defined in
+libxdp. This allows the user to add, delete, update, prioritize, and
+re-prioritize the multiple programs on each interface. There is also support to
+configure whether the flow of execution should terminate and return or continue
+to the next program in the list based on the return value.
+
+While TC natively supports multiple programs on each attach point, it lacks the
+controls and flexibility enabled by the multi-prog protocol. bpfd therefore also
+supports the same XDP multi-prog solution for TC programs which has the added
+benefit of a consistent user experience for both XDP and TC programs.
+
+eBPF programs are also difficult to debug on a system.  The visibility provided
+by bpfd can be a key tool in understanding what is deployed and how they may
+interact.
+
+[Aya]:https://aya-rs.dev/
+[multi-prog]:https://github.com/xdp-project/xdp-tools/blob/master/lib/libxdp/protocol.org
+
+
+### bpfd Kubernetes Support
+
+The benefits of bpfd are brought to Kubernetes by the bpfd operator.  The bpfd
+operator is developed in Go using the Operator SDK framework, so it should be
+familiar to most Kubernetes application developers. The bpfd operator deploys a
+daemonset, containing both bpfd and the bpfd agent processes on each node.
+Rather than making requests directly to bpfd with the gRPC API or bpfctl as
+described above, Kubernetes applications use bpfd custom resource definitions
+(CRDs) to make requests to bpfd to load and attach eBPF programs.  bpfd uses two
+types of CRDs; Program CRDs for each eBPF program type (referred to as *Program
+CRDs, where * = Xdp, Tc, etc.) created by the application to express the desired
+state of an eBPF program on the cluster, and per node BpfProgram CRDs created by
+the bpfd agent to report the current state of the eBPF program on each node.
+
+Using XDP as an example, the application can request that an XDP program be
+loaded on multiple nodes using the XdpProgram CRD, which includes the necessary
+information such as the bytecode image to load, interface to attach it to, and
+priority.  An XdpProgram CRD that would do the same thing as the bpfctl command
+shown above on every node in a cluster is shown below.
+
+```yaml
+apiVersion: bpfd.dev/v1alpha1
+kind: XdpProgram
+metadata:
+  labels:
+    app.kubernetes.io/name: xdpprogram
+  name: xdp-pass-all-nodes
+spec:
+  sectionname: pass
+  # Select all nodes
+  nodeselector: {}
+  interfaceselector:
+    primarynodeinterface: true
+  priority: 0
+  bytecode:
+    image:
+      url: quay.io/bpfd-bytecode/xdp_pass:latest
+  globaldata:
+    GLOBAL_u8:
+      - 0x01
+```
+
+The bpfd agent on each node watches for the *Program CRDs, and makes calls to
```

Comment:
The `*Program`, the * is interpreted as italics until the next * in this paragraph. Not sure why it didn't do it above, but need to figure out how to escape it.

Reply:
I think I fixed it.  We'll see how it renders on netlify.